version1/06 - Access control.md

```diff
@@ -0,0 +1,217 @@
+
+---
+
+# Access control diagram
+
+![Access groups diagram](img/access-groups.png)
+
+# Access control explained
+
+The Documaster mandatory access control relies on access groups (defined in Documaster) in order to configure user access to the system. Users are assigned to access groups dynamically based on the user claims returned by the connected OpenID Connect identity provider. As a result, the Documaster access control is tightly coupled to the group membership setup in the configured OpenID Connect identity provider as each access group in Documaster is configured to map to one or more groups in the OpenID Connect identity provider.
+
+By default, when creating a new Access group, that access group will have no access to the data in Documaster. In order for data to be readable, editable, or deletable by the group, "Explicit permissions" must be assigned to the desired entity types. Permissions are "inherited" meaning that the combination of the `ReadThis` and `Read` permission on a resource will make that resource __and all of its children__ (as described in the "Model") readable. The same applies to non-read permissions, such as `Create`, `Update`, etc.
+
+To put this into perspective, in the diagram above:
+* John Doe is a member of both "Administrators" and "Managers" in the OIDC IDP
+* Jane Doe is a member of "Managers" in the OIDC IDP
+* Baby Doe is a member of "Users" in the OIDC IDP
+
+With the above (external to Documaster) configuration in place, one can create the following three access groups in Documaster:
+* Administrators
+* Managers
+* Users
+
+Based on the "claims" attributes of these access groups and considering the fact that the OIDC IDP returns information about the group membership to Documaster, Documaster is able to map the Documaster-based and the IDP-based groups together and grant the described permissions to each user.
+
+# Access Control Types
+
+## Access Group
+
+An AccessGroup is identified by the following attributes:
+
+| Attribute          | Data type    | Comment                                                                                |
+|--------------------|--------------|----------------------------------------------------------------------------------------|
+| id                 | long         | Unique identifier of the access group                                                  |
+| name               | string       | Human-readable name of the access group                                                |
+| description        | string       | Description of the access group                                                        |
+| claims             | string array | External user claims upon which a user will be assigned as a member of an access group |
+| globalPermissions  | string array | [Global permissions](#explicit-permissions)                                            |
```

Comment:
Were explicit permissions meant to be used here? Edit: I suppose a new endpoint for assigning global permissions will be added to v1?